src/Features/CSharp/Portable/AssignOutParameters/AssignOutParametersAboveReturnCodeFixProvider.cs

```diff
@@ -18,6 +18,11 @@ namespace Microsoft.CodeAnalysis.CSharp.AssignOutParameters
     [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
     internal class AssignOutParametersAboveReturnCodeFixProvider : AbstractAssignOutParametersCodeFixProvider
     {
+        [ImportingConstructor]
+        public AssignOutParametersAboveReturnCodeFixProvider()
+        {
+        }
```

Comment:
i'm not really getting why this is any better.  Should we not just fixup the check to say "if you don't have an implicit public constructor, then you must have at least one constructor that has an `ImportingConstructor`?"

Reply:
RS0035 (which I am not enabling as part of _this_ pull request) requires the importing constructor be marked `[Obsolete]` to prevent construction outside of the MEF container. It is not possible to do this without an explicit constructor.

Reply:
Ah, ok.  in that case i'm ok with this.